src/goto-instrument/contracts/contracts.cpp

```diff
@@ -199,7 +226,23 @@ void code_contractst::check_apply_loop_contracts(
 
   // Now havoc at the loop head.
   // Use insert_before_swap to preserve jumps to loop head.
-  goto_function.body.insert_before_swap(loop_head, havoc_code);
+  insert_before_swap_and_advance(goto_function.body, loop_head, havoc_code);
+
+  // Generate: assignments to store the multidimensional decreases clause's
+  // value before the loop
+  if(!decreases_clause.is_nil())
+  {
+    for(size_t i = 0; i < old_decreases_vars.size(); i++)
```

Comment:
updated previous loops with `size_t` as well for consistency.

Reply:
Fixed as suggested.